### PR TITLE
Add org.openttd.OpenTTD

### PR DESCRIPTION
**Pull request by ghisvail**
My second Flatpak after Zulip. This one requires quite a few dependencies, some of which might be worth contributing to `shared-modules`? Looking forward to your feedback.

### CODE REVIEW

**Comment by ghisvail** [comment]
screenshot from 2018-04-23 09-09-16

File: org.openttd.OpenTTD.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  "branch": "stable",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "openttd",
+  "rename-icon": "openttd",
+  "rename-desktop-file": "openttd.desktop",
+  "finish-args": [
+    "--device=dri",
+    "--filesystem=home",
```
Comment (barthalion) [inline org.openttd.OpenTTD.json]: Does it need home access?

File: org.openttd.OpenTTD.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.openttd.OpenTTD.appdata.xml]: Have you tried contributing this file upstream?

**Comment by barthalion** [comment]
bot, build org.openttd.OpenTTD

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "shared-modules"]
+	path = shared-modules
+	url = git@github.com:flathub/shared-modules.git
```
Comment (barthalion) [inline .gitmodules]: Please use https here.

File: org.openttd.OpenTTD.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://media.openttd.org/images/screens/1.4/02-opengfx-1920x1200.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://www.openttd.org/</url>
+  <developer_name>The OpenTTD team</developer_name>
+  <update_contact>info@openttd.org</update_contact>
+  <releases>
```
Comment (barthalion) [inline org.openttd.OpenTTD.appdata.xml]: One more, appdata misses [OARS](https://hughsie.github.io/oars/) section.

File: org.openttd.OpenTTD.json (hunk 1 above)
Comment (ghisvail) [reply org.openttd.OpenTTD.json]: For exporting save files and additional music, afaik.

File: org.openttd.OpenTTD.appdata.xml (hunk 2 above)
Comment (ghisvail) [reply org.openttd.OpenTTD.appdata.xml]: They have been pinged, see Github link ^.

File: org.openttd.OpenTTD.json [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      "buildsystem": "simple",
+      "build-commands": [
+        "tar -xf opengfx-0.5.2.tar",
+        "mkdir -p /app/share/games/openttd/baseset/opengfx",
+        "cp -r opengfx*/* /app/share/games/openttd/baseset/opengfx"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://binaries.openttd.org/extra/opengfx/0.5.2/opengfx-0.5.2-all.zip",
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply org.openttd.OpenTTD.json]: Indeed. You can either use these, or the proprietary assets from the original TTD game.

**Comment by nedrichards** [comment]
bot, build org.openttd.OpenTTD

**Comment by nedrichards** [comment]
The icu build fails on arm: https://flathub.org/builds/#/builders/47/builds/84

**Comment by ghisvail** [comment]
> The icu build fails on arm

Not sure what to do. I could try a more recent version?

**Comment by nedrichards** [comment]
bot, build org.openttd.OpenTTD

**Comment by nedrichards** [comment]
Hrm, still fails. https://flathub.org/builds/#/builders/49/builds/88 looks like debian etc. work around this by disabling libicudata, but I'm not sure of the implications of that. Just add a flatpak.json file that disables building on arm for now and it looks ready to merge to me.

**Comment by nedrichards** [comment]
Oh, actually it fails to build on i386 and x86-64 as well now: https://flathub.org/builds/#/builders/2/builds/1010

revert the icu change!

**Comment by barthalion** [comment]
Try with this patch: https://github.com/archlinuxarm/PKGBUILDs/blob/master/core/icu/icudata-stdlibs.patch

**Comment by barthalion** [comment]
Also to fix it with latest ICU for all arches, you need to add `-DU_USING_ICU_NAMESPACE=1` to CXXFLAGS.

**Comment by ghisvail** [comment]
I backtracked to ICU v57.1 which is the one currently used in Debian stable, plus the patch mentioned above. It builds fine on my machine.

**Comment by barthalion** [comment]
bot, build org.openttd.OpenTTD

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.openttd.OpenTTD

**Comment by barthalion** [comment]
@ghisvail They probably won't be interested, but could you let upstream developers know that there is a flathub repository that they can get write access to if they would like to?

Thanks!

**Comment by ghisvail** [comment]
@barthalion Done, forwarded [here](https://github.com/OpenTTD/OpenTTD/issues/6778).

**Comment by barthalion** [comment]
That's perfect, kudos.

**Comment by ghisvail** [comment]
@barthalion Actually, they would welcome a PR for it. What would be the procedure then?

Upstream already provides a `.desktop` file [here](https://github.com/OpenTTD/OpenTTD/tree/master/media). Do I just dump the AppStream metadata there and ask them to install the file under `/usr/share/metainfo`, or are there more steps involved to complete the integration?

**Comment by nedrichards** [comment]
Sweet! So it depends on a couple of bits. Firstly, how do they localize? You'll want to put the appdata in such a way that it can get translations (looks like they've got that setup for the desktop file as well).

Secondly - how do they want to add the release data? If it's just a number they can probably hook up autopopulation with the build script. If they want more detailed release notes in there then it'll likely have to be manual commits.

Also `$PREFIX/share/metainfo` ;-)

But this is great!

**Pull request closed without merging**